lib/ClangImporter/ClangImporter.cpp

```diff
@@ -984,6 +984,21 @@ ClangImporter::getClangArguments(ASTContext &ctx) {
   return invocationArgStrs;
 }
 
+/// Gather all -Xclang flags from a Clang driver command-line
+static void extractClangFrontendInvocationArgs(const ArrayRef<std::string> invocationArgStrs,
+                                               std::vector<const char *> &clangFrontendArgs) {
+  clangFrontendArgs.push_back(invocationArgStrs.front().c_str());
+  clangFrontendArgs.push_back("-cc1");
+  auto argIter = invocationArgStrs.begin();
+  while ((argIter = std::find_if(argIter, invocationArgStrs.end(),
+                                 [](std::string arg) {
+                                   return arg == "-Xclang";
```

Comment:
Technically, `-Xclang` is a driver argument. I think it's counter-intuitive having to specify driver arguments when trying to _avoid_ running the driver. Can we make this work with just `-Xcc <frontend-arg>`? The presence of `-direct-clang-cc1-module-build` is already telling Swift you're specifying frontend arguments instead of driver arguments. WDYT?

Reply:
Yes, I like this suggestion. I refactored this to assume that `-Xcc` arguments are meant for the frontend if `-direct-clang-cc1-module-build` is present. This also simplifies the logic quite a bit since we can just use `invocationArgStrs` directly in that case.